src/core/shape_inference/include/group_normalization_shape_inference.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <array>
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/group_normalization.hpp>
+
+namespace ov {
+namespace op {
+namespace v12 {
+template <class TShape>
+std::vector<TShape> shape_infer(const GroupNormalization* op,
+                                const std::vector<TShape>& input_shapes,
+                                const std::map<size_t, HostTensorPtr>& constant_data = {}) {
+    const auto& inputs_count = input_shapes.size();
+    NODE_VALIDATION_CHECK(op, (inputs_count == 3));
+    TShape output_shape;
+
+    const auto data_partial_shape = input_shapes[0];
+    const auto data_rank = data_partial_shape.rank();
+    const auto scale_partial_shape = input_shapes[1];
+    const auto bias_partial_shape = input_shapes[2];
```

Comment:
```suggestion
    const auto& data_partial_shape = input_shapes[0];
    const auto data_rank = data_partial_shape.rank();
    const auto& scale_partial_shape = input_shapes[1];
    const auto& bias_partial_shape = input_shapes[2];
```
to avoid copies